connectors/src/connectors/slack/bot.ts

```diff
@@ -0,0 +1,188 @@
+import { literal } from "sequelize";
+
+import { ChatSessionUpdateEvent, DustAPI } from "@connectors/lib/dust_api";
+import {
+  Connector,
+  SlackChatBotMessage,
+  SlackConfiguration,
+} from "@connectors/lib/models";
+import { Err, Ok, Result } from "@connectors/lib/result";
+import logger from "@connectors/logger/logger";
+
+import {
+  getAccessToken,
+  getSlackClient,
+  getUserName,
+  whoAmI,
+} from "./temporal/activities";
+
+export async function botAnswerMessageWithErrorHandling(
+  message: string,
+  slackTeamId: string,
+  slackChannel: string,
+  slackUserId: string,
+  slackMessageTs: string
+): Promise<Result<ChatSessionUpdateEvent, Error>> {
+  const slackConfigurations = await SlackConfiguration.findAll({
+    where: {
+      slackTeamId: slackTeamId,
+    },
+  });
+
+  if (slackConfigurations.length === 0) {
+    return new Err(
+      new Error(
+        `Failed to find slack configuration for Slack team id: ${slackTeamId}`
+      )
+    );
+  }
+  const slackConfig = slackConfigurations[0];
```

Comment:
This probably deserve a small comment to explain the fact that we randomly pick the first one. Can simply link to this PR discussion